stdlib/LinearAlgebra/src/svd.jl

```diff
@@ -92,18 +97,25 @@ julia> A
   0.0       0.0  -2.0  0.0  0.0
 ```
 """
-function svd!(A::StridedMatrix{T}; full::Bool = false) where T<:BlasFloat
+function svd!(A::StridedMatrix{T}; full::Bool = false, alg::Algorithm = default_svd_alg(A)) where T<:BlasFloat
     m,n = size(A)
     if m == 0 || n == 0
         u,s,vt = (Matrix{T}(I, m, full ? m : n), real(zeros(T,0)), Matrix{T}(I, n, n))
     else
-        u,s,vt = LAPACK.gesdd!(full ? 'A' : 'S', A)
+        if typeof(alg) == DivideAndConquer
```

Comment:
Better to use dispatch for this

Reply:
```julia
function svd!(A::StridedMatrix{T}; full::Bool = false, alg::Algorithm = default_svd_alg(A)) where T<:BlasFloat
    m,n = size(A)
    if m == 0 || n == 0
        u,s,vt = (Matrix{T}(I, m, full ? m : n), real(zeros(T,0)), Matrix{T}(I, n, n))
    else
        u,s,vt = _lapack_svd(A,full,alg)
    end
    SVD(u,s,vt)
end


_lapack_svd(A::StridedMatrix{T}, full::Bool, alg::Algorithm) where T<:BlasFloat = throw(ArgumentError("Unsupported value for `alg` keyword."))
_lapack_svd(A::StridedMatrix{T}, full::Bool, alg::DivideAndConquer) where T<:BlasFloat = LAPACK.gesdd!(full ? 'A' : 'S', A)
function _lapack_svd(A::StridedMatrix{T}, full::Bool, alg::QRIteration) where T<:BlasFloat
    c = full ? 'A' : 'S'
    u,s,vt = LAPACK.gesvd!(c, c, A)
end
```

Is this what you have in mind?

Reply:
Yes exactly. (I'd just call it `_svd!`, though)